yarn-project/noir-contracts/src/contracts/schnorr_auth_witness_account_contract/src/main.nr

```diff
@@ -19,23 +68,62 @@ contract SchnorrAuthWitnessAccount {
           payload.serialize(), 
           GENERATOR_INDEX__SIGNATURE_PAYLOAD
         )[0];
-        _inner_is_valid(message_hash, context.this_address());
+        assert(_inner_is_valid(message_hash, context.this_address()));
         payload.execute_calls(&mut context);
     }
 
     #[aztec(private)]
     fn is_valid(
         message_hash: Field
     ) -> Field {
-        _inner_is_valid(message_hash, context.this_address());
-        0xe86ab4ff
+        if (_inner_is_valid(message_hash, context.this_address())){
+            0xe86ab4ff
+        } else {
+            0
+        }
+    }
+
+    #[aztec(public)]
+    fn is_valid_public(
+        message_hash: Field,
+    ) -> Field {
+        let storage = Storage::init(Option::none(), Option::some(&mut context));
+        let value = storage.approved_action.at(message_hash).read();
+        if (value == 1){
+            0xe86ab4ff
+        } else {
+            0
+        }
+    }
+
+    #[aztec(private)]
+    fn set_is_valid_storage(
```

Comment:
We haven't even launched and have already found a use case for transient storage